### DISCUSSION TITLE
How can I enforce the static build of a dependency in a manifest file?

### DISCUSSION THREAD

**Discussion by McKillroy** [microsoft/vcpkg#23881] [Q&A] [answered]
I want to build a library which also depends on other libraries using vcpkg manifest mode inside a custom registry.
How can I enforce, that these dependencies are built statically (x64-windows-static or x64-linux-static triplets) in the manifest file?
From the docs using dependency objects seems not to support this (at least I know of no way to do this).

example manifest:
```
{
    "name": "mylib",
    "version-semver": "0.0.1",
    "dependencies":[
        "libsodium",
        "mbedtls"
    ]
}
```

**Comment by McKillroy** [comment]
I finally came to a sort of hackish workaround like this at the beginning of my cmake lists file, before any project statement:
```cmake
if(WIN32)
    SET(VCPKG_TARGET_TRIPLET "x64-windows-static")
elseif(UNIX)
    SET(VCPKG_TARGET_TRIPLET "x64-linux")
else()    
    message("==================================================================")
    message("Unsupported platform: neither Windows or Linux")
    message("==================================================================")
    message(FATAL_ERROR )
endif()

SET(BUILD_SHARED_LIBS OFF) 
set(VCPKG_CRT_LINKAGE dynamic)
```

Any suggestions how to improve this? I doubt it's the way how things should be done. Is it?

**Comment by Osyotr** [comment, marked as answer]
It's not possible to set linkage in manifest file. You should pass `VCPKG_TARGET_TRIPLET` to cmake (for example, via `CMakePresets.json`).
If you want only certain libraries to be static, you should create an overlay-triplet and make a condition on packages you want to be static.
`x64-windows-mixed.cmake`:
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)

if("${PORT}" STREQUAL "portnameiwanttobestatic")
    set(VCPKG_LIBRARY_LINKAGE static)
endif()
```

See
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#per-port-customization
https://github.com/microsoft/vcpkg/blob/master/docs/examples/overlay-triplets-linux-dynamic.md

**Reply by McKillroy** [reply]
That seems to work. Thanks a lot!
